haystack/classifier/farm.py

```diff
@@ -0,0 +1,282 @@
+import logging
+import multiprocessing
+from pathlib import Path
+from typing import List, Optional, Union
+
+from farm.data_handler.data_silo import DataSilo
+from farm.data_handler.processor import TextClassificationProcessor
+from farm.infer import Inferencer
+from farm.modeling.optimization import initialize_optimizer
+from farm.train import Trainer
+from farm.utils import set_all_seeds, initialize_device_settings
+
+from haystack import Document
+from haystack.classifier.base import BaseClassifier
+
+logger = logging.getLogger(__name__)
+
+
+class FARMClassifier(BaseClassifier):
+    """
+    This node classifies documents and adds the output from the classification step to the document's meta data.
+    The meta field of the document is a dictionary with the following format:
+    'meta': {'name': '450_Baelor.txt', 'classification': {'label': 'neutral', 'probability' = 0.9997646, ...} }
+
+    |  With a FARMClassifier, you can:
+     - directly get predictions via predict()
+     - fine-tune the model on text classification training data via train()
+
+    Usage example:
+    ...
+    retriever = ElasticsearchRetriever(document_store=document_store)
+    classifier = FARMClassifier(model_name_or_path="deepset/bert-base-german-cased-sentiment-Germeval17")
+    p = Pipeline()
+    p.add_node(component=retriever, name="ESRetriever", inputs=["Query"])
+    p.add_node(component=classifier, name="Classifier", inputs=["ESRetriever"])
+
```

Comment:
Would it actually also work to use this node in an indexing pipeline? I mean something like FileConverter->Preprocessor->Classifier->DocStore

So we would basically append meta data to the docs at indexing time...

Reply:
Doesn't have to be part of this PR if it requires bigger changes, but maybe you can document what's missing for that use case and create a separate issue

Reply:
created an issue here: https://github.com/deepset-ai/haystack/issues/1281